core/domain/question_services.py

```diff
@@ -85,31 +86,113 @@ def _create_new_question(committer_id, question, commit_message):
     create_question_summary(question.id, committer_id)
 
 
-def create_new_question_skill_link(question_id, skill_id, skill_difficulty):
-    """Creates a new QuestionSkillLink model.
+def create_multi_question_skill_links_for_question(
+        user_id, question_id, skill_ids, skill_difficulties):
+    """Creates multiple new QuestionSkillLink models. Also adds
+    the skill ids to the linked skill ids of the Question model.
 
     Args:
+        user_id: str. ID of the creator.
+        question_id: str. ID of the question linked to the skills.
+        skill_ids: list(str). ID of the skills to which the question is linked.
+        skill_difficulties: list(float). The difficulty of the question with
+            respect to a skill, represented by a float between
+            0 and 1 (inclusive).
+
+    Raises:
+        Exception. Skill difficulties and skill ids should match. The lengths of
+        the two lists are different.
+    """
+    if len(skill_ids) != len(skill_difficulties):
+        raise Exception(
+            'Skill difficulties and skill ids should match. The lengths of the '
+            'two lists are different.')
+    question = get_question_by_id(question_id)
+
+    new_question_skill_link_models = []
+    for index, skill_id in enumerate(skill_ids):
+        new_question_skill_link_models.append(
+            question_models.QuestionSkillLinkModel.create(
+                question_id, skill_id, skill_difficulties[index]))
+
+    new_linked_skill_ids = copy.deepcopy(question.linked_skill_ids)
+    new_linked_skill_ids.extend(skill_ids)
+
+    _update_linked_skill_ids_of_question(
+        user_id, question_id, new_linked_skill_ids,
+        question.linked_skill_ids)
+    question_models.QuestionSkillLinkModel.put_multi_question_skill_links(
+        new_question_skill_link_models)
+
+
+def create_new_question_skill_link(
+        user_id, question_id, skill_id, skill_difficulty):
+    """Creates a new QuestionSkillLink model and adds the skill id
+    to the linked skill ids for the Question model.
+
+    Args:
+        user_id: str. ID of the creator.
         question_id: str. ID of the question linked to the skill.
         skill_id: str. ID of the skill to which the question is linked.
         skill_difficulty: float. The difficulty between [0, 1] of the skill.
     """
+    question = get_question_by_id(question_id)
+
     question_skill_link_model = question_models.QuestionSkillLinkModel.create(
         question_id, skill_id, skill_difficulty)
+
+
+    new_linked_skill_ids = copy.deepcopy(question.linked_skill_ids)
+    new_linked_skill_ids.append(skill_id)
```

Comment:
Do we need to check that skill_id isn't in the list already? Should test this case too.

Reply:
Done